plantcv/plantcv/photosynthesis/analyze_yii.py

```diff
@@ -0,0 +1,165 @@
+# Fluorescence Analysis (Fv/Fm parameter)
+
+import os
+import cv2
+import numpy as np
+import pandas as pd
+from plotnine import ggplot, aes, geom_line, geom_label
+from plantcv.plantcv import fatal_error
+from plantcv.plantcv._debug import _debug
+from plantcv.plantcv import params
+from plantcv.plantcv import outputs
+
+
+def analyze_yii(ps, mask, measurement, bins=256, label="default"):
+    """Calculate and analyze PSII efficiency estimates from fluorescence image data.
+
+    Inputs:
+    ps          = photosynthesis xarray DataArray
+    mask        = mask of plant (binary, single channel)
+    measurement = choose which measurement routine to analyze: "Fv/Fm", "Fv'/Fm'", or "Fq'/Fm'"
+    bins        = number of bins (1 to 256 for 8-bit; 1 to 65,536 for 16-bit; default is 256)
+    label       = optional label parameter, modifies the variable name of observations recorded
+
+    Returns:
+    ind_fig   = Fluorescence induction curve plot
+    hist_fig  = Histogram of efficiency estimate
+    yii_img   = Image of efficiency estimate values
+
+    :param ps: xarray.core.dataarray.DataArray
+    :param mask: numpy.ndarray
+    :param measurement: str
+    :param bins: int
+    :param label: str
+    :return ind_fig: plotnine.ggplot.ggplot
+    :return hist_fig: plotnine.ggplot.ggplot
+    :return yii_img: numpy.ndarray
+    """
+
+    prot_vars = {
+        "fv/fm": {"Fm": "Fm", "F0": "F0"},
+        "fv'/fm'": {"Fm": "Fm", "F0": "F0"},
+        "fq'/fm'": {"Fm": "Fm'", "F0": "F'"}
+    }
+
+    if measurement.lower() not in prot_vars:
+        fatal_error(f"Measurement {measurement} is not one of Fv/Fm, Fv'/Fm', or Fq'/Fm'")
+
+    # Analyze fluorescence induction curve to identify max fluorescence
+    ind_df = _fluor_induction(ps=ps, mask=mask, measurement=measurement)
+
+    # Make fluorescence induction curve figure using plotnine
+    ind_fig = (ggplot(data=ind_df, mapping=aes(x="Timepoints", y="Fluorescence", color="Measurement"))
+               + geom_line(show_legend=True))
+
+    # Select the F0, F0', or F' frame
+    fmin = ps.sel(frame_label=ps.attrs[prot_vars[measurement.lower()]["F0"]]).data
```

Comment:
if ps has another dimension `parameter` then we can do something like 
```
def calc_yii(da):
    return (da.sel(frame='Fmp') - da.sel(frame='Fp'))/da.sel(frame='Fmp')

yii = ps.groupby('parameter', squeeze=True).map(calc_yii).drop_vars('frame_label')
```
i think we still need to map each F0,... to a Fm' and F' with a dict. maybe that can be an input to calc_yii

Reply:
```
ps_expanded = ps.expand_dims(dim = 'measurement_num', axis=3)
def calc_yii(da, fp, fmp):
    return (da.sel(frame_label=fmp) - da.sel(frame_label=fp))/da.sel(frame_label=fmp)
fvfm = ps_expanded.groupby('measurement_num', squeeze=True).map(calc_yii,args = ('F0', 'F5')).drop('frame_label')
```
replace F0 and F5 with the dict inferno

Reply:
woops, i double checked this and it doesn't give what I expected. will need to investigate another day

Reply:
assign new coord to existing dimension
http://xarray.pydata.org/en/stable/generated/xarray.DataArray.assign_coords.html?highlight=add%20coords